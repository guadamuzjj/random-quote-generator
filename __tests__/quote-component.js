import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Quote from '../components/quote';

describe('Quote component', () => {
  const testQuote = {
    quoteText: 'Every science begins as philosophy and ends as art.',
    quoteAuthor: 'Will Durant',
    quoteGenre: 'science',
  }

  it('should be rendered', () => {
    const { getByTestId } = render(<Quote info={testQuote} />);
    expect(getByTestId('quoteItem')).toBeDefined();
  });

  it('should render a skeleton if is not defined yet', () => {
    const { getByTestId } = render(
      <Quote info={null} />
    );

    expect(getByTestId('quoteSkeleton')).toBeInTheDocument();
  });

  it('should render with the featured info', () => {
    const { getByTestId } = render(
      <Quote
        featured
        info={testQuote}
      />
    );

    expect(getByTestId('quoteText')).toHaveTextContent(testQuote.quoteText);
    expect(getByTestId('authorLink')).toHaveAttribute('href', `/authors/${encodeURI(testQuote.quoteAuthor)}`);
    expect(getByTestId('quoteAuthor')).toHaveTextContent(testQuote.quoteAuthor);
    expect(getByTestId('quoteGenre')).toHaveTextContent(testQuote.quoteGenre);
  });


  it('should render with the basic info', () => {
    const { queryByTestId } = render(
      <Quote info={testQuote} />
    );

    expect(queryByTestId('quoteText')).toHaveTextContent(testQuote.quoteText);
    expect(queryByTestId('quoteAuthor')).toBeNull();
    expect(queryByTestId('quoteGenre')).toBeNull();
  });

  it('should render a hyphen as genre if genre is not provided', () => {
    const { quoteText, quoteAuthor } = testQuote;

    const { getByTestId } = render(
      <Quote
        featured
        info={quoteText, quoteAuthor}
      />
    );
    expect(getByTestId('quoteGenre')).toHaveTextContent('-');
  });

  it('match to snapshot', () => {
    const quoteComponent = renderer.create(<Quote featured info={testQuote} />).toJSON();
    expect(quoteComponent).toMatchSnapshot();
  });
});
