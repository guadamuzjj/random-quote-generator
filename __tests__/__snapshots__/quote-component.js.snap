// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component match to snapshot 1`] = `
<div
  data-testid="quoteItem"
>
  <div
    className="border-l-8 border-yellow-500"
  >
    <div
      className="pl-8 sm:pl-16 xl:pl-24"
    >
      <h1
        className="text-lg md:text-xl lg:text-2xl xl:text-3xl"
        data-testid="quoteText"
      >
        Every science begins as philosophy and ends as art.
      </h1>
    </div>
  </div>
  <div
    className="border-l-8 border-transparent"
  >
    <a
      data-testid="authorLink"
      href="/authors/Will%20Durant"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        className="flex justify-between items-center px-8 sm:pl-16 xl:pl-24 py-4 mt-12 text-gray-700 group hover:bg-gray-800 hover:text-white"
      >
        <div
          className="flex flex-col"
        >
          <span
            className="text-base md:text-lg lg:text-xl xl:text-xl font-bold"
            data-testid="quoteAuthor"
          >
            Will Durant
          </span>
          <span
            className="text-sm font-medium"
            data-testid="quoteGenre"
          >
            science
          </span>
        </div>
        <div
          className="hidden group-hover:block"
        >
          <svg
            fill="#fff"
            height={24}
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.025 1l-2.847 2.828 6.176 6.176h-16.354v3.992h16.354l-6.176 6.176 2.847 2.828 10.975-11z"
            />
          </svg>
        </div>
      </div>
    </a>
  </div>
</div>
`;
